src/Microsoft.AspNetCore.Blazor/RenderTree/RenderTreeDiffBuilder.cs

```diff
@@ -179,8 +179,8 @@ public static void DisposeFrames(RenderBatchBuilder batchBuilder, ArrayRange<Ren
                     // These two attributes have the same sequence and name. Keep merging.
                     AppendDiffEntriesForAttributeFrame(ref diffContext, oldStartIndex, newStartIndex);
 
-                    oldStartIndex = NextSiblingIndex(oldTree[oldStartIndex], oldStartIndex);
-                    newStartIndex = NextSiblingIndex(newTree[newStartIndex], newStartIndex);
+                    oldStartIndex++;
+                    newStartIndex++;
```

Comment:
Should always have just been "++" here, since attribute frames are always a continuous flat list. Attributes don't have descendants to skip over.